<!doctype html>
<html>
	<head>
		<meta charset="utf-8">
		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

		<title>Developing & benchmarking methods for analysing transcriptomics data</title>

		<link rel="stylesheet" href="css/reset.css">
		<link rel="stylesheet" href="css/reveal.css">
		<link rel="stylesheet" href="css/theme/solarized.css">
    <link rel="stylesheet" href="css/my_theme.css">
    <link rel="stylesheet" href="fonts/fontawesome.css">

		<!-- Theme used for syntax highlighting of code -->
		<link rel="stylesheet" href="lib/css/monokai.css">

		<!-- Printing and PDF exports -->
		<script>
			var link = document.createElement( 'link' );
			link.rel = 'stylesheet';
			link.type = 'text/css';
			link.href = window.location.search.match( /print-pdf/gi ) ? 'css/print/pdf.css' : 'css/print/paper.css';
			document.getElementsByTagName( 'head' )[0].appendChild( link );
    </script>
	</head>
	<body>
		<div class="reveal">
			<div class="slides">
				<section id="hello" data-timing="40">
          <aside class="notes">
            Hello [...]
          </aside>

					<div class="chapter-content">
            <div class="container">
              <div class="col bordered full">
                <div class="container-row">
                  <div class="row">
                    <div>
                      <sup><i class="fas fa-graduation-cap"></i></sup>
                      <h1>Closed PhD defence</h1>
                      Wouter Saelens
                    </div>
                  </div>
                  <div class="row">
                    <div>This presentation:<br> <a href="https://zouter.github.io/closed-defence">zouter.github.io/closed-defence
                        <sup><i class="fas fa-external-link-alt" style="font-size:0.7em"></i></sup></a> </div>
                  </div>
                </div>
              </div>
              <div class="col bordered full">
                <div id="overview" data-svg="slides/title/overview.svg"></div>
              </div>
            </div>
          </div>
        </section>
        

				<section id="big-questions" data-timing="56">
          <aside class="notes">
            Why do we even study transcriptomics? Well we have these large questions, such as [...] But to answer them we have to go
            smaller and smaller: look into what cells are doing, which leads us to trying to understand what is happening in this cell. Of course a bunch of molecules, and one molecule that is relatively easy to quantify and which gives us a lot of
            information are the mRNA transcripts. They of course encode for proteins, and their abundances can tell us a lot about
            what the cell is doing.
          </aside>
          <div class="chapter-heading">
            Big questions often have tiny answers
          </div>
          <div class="chapter-content" style="height:500px;">
            <div data-svg="slides/intro/intro.svg" class="full"></div>
          </div>
        </section>

        <section id="intro" data-timing="40">
          <aside class="notes">
            Transcriptomics data is highly dimensional, i.e. a couple to thousands of samples with information about all protein coding genes data, and this requires computational tools. Some of those are listed here, starting from the simpler [...] to the more complex [...].

            The goal in my PhD wasn't set in stone from the beginning. Instead I tried to analyse some data in the department and then see where I could contribute new methods, or compare existing methods.
          </aside>

          <div class="container">
            <a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a
              class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a
              class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a
              class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a>
            <div class="col bordered">
              <div data-svg="slides/intro/tools.svg" class="full"></div>
            </div>
            <div class="col bordered" style="line-height:2em;" data-index="goals">
              <div>
                <p>My goals: </p>
                <p>
                Analyse transcriptome of <br> macrophages and dendritic cells
                </p>
                <p>
                  But...
                </p>
                <ul class="fa-ul" style="list-style-type: none;padding-left:40px">
                  <li>
                    <span class="fa-li"><i class="far fa-square"></i></span>
                    <p>How to do differential expression for 3 conditions?</p>
                  </li>
                  <li>
                    <span class="fa-li"><i class="far fa-square"></i></span>
                    <p>So many module detection methods, which one should I use?</p>
                  </li>
                  <li>
                    <span class="fa-li"><i class="far fa-square"></i></span>
                    <p>So many trajectory inference detection methods, which one should I use?</p>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </section>

        <section data-vertical-align-top id="triwise" data-timing="230">
          <aside class="notes">
            One problem that I encountered early in my PhD is the analysis of 3 biological conditions. When we have two conditions, a pairwise analysis is quite easy. Also if you have more samples, some clustering is typically employed to simplify the dataset and visualize it. The problem is, when I had three conditions, the first approach would require you to do many pairwise comparisons making it difficult to interpret. The other approach would also not work that well because the data is too simple to really try to cluster the genes.

            <hr>

            So that's why I developed this visualisation technique where I convert the data to two dimensions using a barycentric transformation. Genes that are not different in any of the three samples are in the middle. Other genes move towards one of the three poles depending on their expression. The distance to the origin is the magnitude of differential expression. So this is the dataset for which I developed this approach, these are three different macrophage progenitor populations, and some of these were more efficient in generating macrophages and so we were wondering why this was. 

            <br>
            So one of the strengths of this analysis is that you can highlight functional gene sets and see whether they are differentialle expressed in a particular direction. For example, here you can see [...]

            <hr>
            Overall, this approach is interesting in that you can use it for many designs, such as the case where you have one reference and two populations of interest. Depending on the biological question, you also have different tests.
            I'm happy with the output that we have with this project, although the approach also has some issues, such as its limitation to three dimensions and the slow implementation of the permutation test. Finally, I think in the future the package could benefit from having a more theoretical statistical background, for example for the self-contained test.
          </aside>

          <a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a>
          <div class="chapter-heading">
            <span class="chapter-number" style="background-color:#FF4136">2</span> 
            Differential expression for three conditions | 
            <span class="chapter-subheading"  style="color:#FF4136">
              <span data-index="problem">
                  problem
              </span>
              <span data-index="solution">
                    solution
              </span>
              <span data-index="conclusion">
                  conclusion
              </span>
            </span>
          </div>

          <div class="chapter-content" data-index="problem">
            <div class = "container">
              <div class="col bordered" data-index="existing">
                <div>
                  <p>Comparing two populations: <br>easy</p>
                  <p><img src="slides/triwise/two_example.png" style="width:75%"></p>
                </div>
              </div>
              <div class="col bordered" data-index="needed">
                <div>
                  <p style="align-self: flex-start">Comparing three populations: </p>
                  <p style="line-height:300px;">a bit more difficult</p>
                </div>   
              </div>
              <div class="col bordered" data-index="existing">
                <div>
                  <p>Comparing many populations: <br>straightforward</p>
                  <p><img src="slides/triwise/many_example.png" style="width:75%"></p>
                </div>
              </div>
            </div>
          </div>
          <div class="chapter-content" data-index="solution">
            <div class="container">
                <div class="col bordered" data-svg="slides/triwise/lianne.svg" style="flex:2"></div>
                <div class="col bordered" data-svg="slides/triwise/lianne2.svg" style="flex:1" data-index="significant"></div>
            </div>
          </div>

          <div class="chapter-content" data-index="conclusion">
            <div class="container-row conclusion-container">
              <div class="row bordered conclusion-box">
                <div class="conclusion-icon"><i class="fas fa-check"></i></div>
                <div class="conclusion-content">
                  <ul>
                    <li>Useful for multiple designs</li>
                    <li>Different tests depending on the biological question</li>
                  </ul>
                </div>
              </div>
              <div class="row bordered conclusion-box">
                <div class="conclusion-icon"><i class="fas fa-newspaper"></i></div>
                <div class="conclusion-content">
                  <ul>
                    <li>3 co-publications</li>
                    <li>4 other studies</li>
                    <li>R package</li>
                  </ul>
                </div>
              </div>
              <div class="row bordered conclusion-box" data-index="future">
                <div class="conclusion-icon"><i class="fas fa-times"></i></div>
                <div class="conclusion-content">
                  <ul>
                    <li>Limited to 3 dimensions</li>
                  </ul>
                </div>
              </div>
            </div>
          </div>


        </section>

        <section data-vertical-align-top id="modbenchmark"  data-timing="216">
          <aside class="notes">
            My next project concerned the use of gene module detection methods. The reason we want to use these methods is that we want to find genes that are functionally and/or regulatory related. This would help us to better understand the process that we are studying, which is often regulated by many genes. The way we do this is that we first group genes based on their similar expression profile, and we assume that these genes because they are co-expressed, will also share function and regulation.

            There are many methods to find these modules, and my goal was to evaluate them. So what I did was to actually test this assumption: are these methods detecting co-regulated or functionally related modules?

            <hr>

            There are many methods to get modules. The most popular one are of course the clustering methods, such as hierarchical clustering shown here. But you also have alternatives such as decomposition and biclustering, which will look for local co-expression patterns and potentially overlapping modules. You also have network inference based methods which will infer some kind of regulatory network together with module detection.

            <br>

            I evaluated these methods on a couple of datasets by checking how well the genes within the modules were known to be co-regulated. While there is some variability between datasets, it's pretty corelated overall. I also checked the running times of course, especially biclustering and network inference methods do not scale very well. You can then average these values to get an overall score for each method, both a training score (which may overfit on a dataset) and a test score. Overall we found that biclustering method surprisingly do not perform well and are quite parameter dependent. The best performers were some of these decomposition methods that provide overlapping and local-co-expression modules with decent scalability, at least on the bulk datasets tested here. There are many more analyses included in the paper delving further into these results.

            <hr>

            To conclude, I'm really happy to have done the first comprehensive evaluation of these methods. We highlight the potential for decomposition methods for these kinds of analyses, but still pinpoint some challenges for users and developers regarding interpretion. With one publication, but being used by several others, I think this was a succes. Of course, during my PhD a lot changed in the transcriptomics field, and some of these results may no longer apply. In the future, it may be useful to repeat this analysis with single-cell data and other omic types.

          </aside>


          <a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a>
          <div class="chapter-heading">
            <span class="chapter-number" style="background-color:#FF851B">3</span> 
            Module detection comparison | 
            <span class="chapter-subheading" style="color:#FF851B">
              <span data-index="problem">
                  problem
              </span>
              <span data-index="solution">
                    solution
              </span>
              <span data-index="conclusion">
                  conclusion
              </span>
            </span>
          </div>
          <div class="container-row chapter-content" data-index="problem">
            <div class="full bordered">
              <div data-svg="slides/modbenchmark/assumption.svg" class="full"></div>
            </div>
          </div>

          <div class="container-row chapter-content" data-index="solution">
            <div class="row bordered">
              <div data-svg="slides/modbenchmark/fig2.svg" class="full"></div>
            </div>
          </div>

          <div class="chapter-content" data-index="conclusion">
            <div class="container-row conclusion-container">
              <div class="row bordered conclusion-box">
                <div class="conclusion-icon"><i class="fas fa-check"></i></div>
                <div class="conclusion-content">
                  <ul>
                    <li>First large-scale evaluation of these methods</li>
                    <li>Decomposition methods for local co-expression</li>
                    <li>Pinpoints challenges to interpretation and accuracy</li>
                  </ul>
                </div>
              </div>
              <div class="row bordered conclusion-box">
                <div class="conclusion-icon"><i class="fas fa-newspaper"></i></div>
                <div class="conclusion-content">
                  <ul>
                    <li>1 first author publication</li>
                    <li>Used by developers & users</li>
                  </ul>
                </div>
              </div>
              <div class="row bordered conclusion-box" data-index="future">
                <div class="conclusion-icon"><i class="fas fa-times"></i></div>
                <div class="conclusion-content">
                  <ul>
                    <li>Not evaluated for single-cell data</li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </section>

        <section data-vertical-align-top id="dynbenchmark" data-timing="300">
          <aside class="notes">

            Another way to look at transcriptomics data is through trajectory inference. In contrast to clustering, where essentially you draw "boundaries" around potential populations, trajectories are continuities in your data. Biologically speaking, these trajectories represent dynamic processes such as cell cycle, differentiation, activation etc.

            During my PhD I saw the rise of these methods, from the first in 2014, onto its "boom" in 2016 and its explosion in 2017.  While we initially wanted to create our own method, after a while we though: screw this, we're going to evaluate them. At one point in 2017 I was literally adding new methods every day just by some simple google scholar searches. The properties of these methods shifted a bit during their growth, first linear, later trees and disconnected trajectories, and now the focus is (or should be) on scalability. As a reference, the growth of these methods lags a bit behind the development of single-cell technologies (obviously).

            <hr>

            During my PhD I together with my dearest colleague, created a evaluation pipeline, and while I could fill hours talking about it, I will just discuss some of the conclusions here. So one of the things I'm most proud of here are the practical guidelines that we created for the users. In essense, one primary determinant in the choice of method is the kind of topology that is expected to be in the data. It's true, this adds some subjectivity, but if it makes your trajectory better, it's perhaps a risk you can take. So in this case for example, if you suspect a tree trajectory to be present in the data, you would follow the flow chart all the way through and ultimately end up with methods such as slingshot. 

            <hr>

            Of course, topology is only one aspect of the data, and there are many more such as dataset size and the prior information that the user has. So to provide context-dependent guidelines, we created this interactive shiny app where a user is given a set of questions and they then are provided with a set of methods. Additional information is also shown, such as indications of instability of the output models.

            <hr>

            So again, this was a first comprehensive evaluation of these methods, but here we really pushed the guidelines to the next level with making them very context-dependent. The project as a whole led to two shared first author papers, one being a review, a couple of preprints, and dozens of R packages some of which are by now on CRAN (as they should).

            Of course, this evaluation also has some limitations, such as its extendability, and the fact that it is already outdated. There is also some discussion in the field, especially from those that developed methods that didn't perform well in the comparison. The whole evaluation pipeline will be reused soon in the form of a competition, and there are still countless of extensions possible from this evaluation, some of which we are doing at the moment.
          
          </aside>

          <a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a
            class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a
            class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a
            class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a>
          <div class="chapter-heading">
            <span class="chapter-number" style="background-color:#93cc2e">4</span>
            Trajectory inference comparison |
            <span class="chapter-subheading" style="color:#93cc2e">
              <span data-index="problem">
                problem
              </span>
              <span data-index="solution">
                solution
              </span>
              <span data-index="conclusion">
                conclusion
              </span>
            </span>
          </div>

          <div class="container-row chapter-content bordered" data-index="problem-1">
              <div class="full" data-svg="slides/dynbenchmark/trajectory_example.svg"></div>
          </div>

          <div class="container-row chapter-content bordered" data-index="problem-2">
              <div class="full" data-svg="slides/dynbenchmark/history.svg"></div>
          </div>
 
        
          <div class="container-row chapter-content bordered" data-index="solution-1">
            <div class="full" data-svg="slides/dynbenchmark/guidelines.svg"></div>
          </div>

          <div class="container-row chapter-content bordered" data-index="solution-2">
            <iframe src="slides/dynbenchmark/dynguidelines.html" style="width:100%;height:100%;max-width:100%;max-height:100%;"></iframe>
          </div>
        
          <div class="chapter-content" data-index="conclusion">
            <div class="container-row conclusion-container">
              <div class="row bordered conclusion-box">
                <div class="conclusion-icon"><i class="fas fa-check"></i></div>
                <div class="conclusion-content">
                  <ul>
                    <li>First large-scale evaluation of these methods</li>
                    <li>Interactive context-dependent guidelines</li>
                  </ul>
                </div>
              </div>
              <div class="row bordered conclusion-box">
                <div class="conclusion-icon"><i class="fas fa-newspaper"></i></div>
                <div class="conclusion-content">
                  <ul>
                    <li>2 shared first-author publications</li>
                    <li>3 preprints</li>
                    <li>Dozens of R packages</li>
                  </ul>
                </div>
              </div>
              <div class="row bordered conclusion-box" data-index="future">
                <div class="conclusion-icon"><i class="fas fa-times"></i></div>
                <div class="conclusion-content">
                  <ul>
                    <li>Already outdated, although still highly relevant</li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        
        
          <aside class="notes">

          </aside>
        </section>


        <section data-vertical-align-top id="conclusions"  data-timing="70">
          <aside class="notes">
            So these was the bulk of the work I did during my PhD, and from those I can derive some overall conclusions. I think the studies showed that benchmarking is really important, because the most popular method is not always the best performing for a user, and good ideas in theory do not always lead to better methods for developers, often simple ideas go really far.

            Moreover, the different metrics we calculated for the methods are often not similar, so you can have a very accurate method but its much less scalable, or vice versa. So finding a balance between these is critical.

            In bio-informatics, there is often a focus on being on top of the ranking, but with both comparisons we showed that this is really problematic thinking. Often, some methods work well on particular use cases, for example particular topologies, and a good benchmark should highlight those and avoid falling in the "one-ranking-that-rules-them-all" idea.

            I already mentioned that the last benchmarking study really has a lot of potential for spin-off studies, and I highlight two of them in the thesis as future perspectives.
          </aside>


          <div class="chapter-heading">
            <span class="chapter-number" style="background-color:darkgray">5</span>
            Overall conclusions
          </div>
          <div class="container-row" style="font-size:0.8em">
            <div class="row bordered" >
              Benchmarking is critical for
              <ul style="list-style: none;text-align:left;">
                <li>
                  → users, because popular ⇏ best performance
                </li>
                <li>
                  → developers, because good ideas ⇏ better for users
                </li>
              </ul>
              </p>
            </div>
            <div class="row bordered">
              Accuracy, scalability, usability and stability are not always correlated. <br>
              Find a balance
            </div>
            <div class="row bordered">
               Performance is context-dependent <br>
               Avoid falling in the "ranking" trap
            </div>
            <div class="row bordered">
              6 primary research articles (2 first author) <br> 3 pre-prints <br> 2 review articles (1 first author) <br> bunch of R packages
            </div>
          </div>
        </section>

        <section data-vertical-align-top id="dyno" data-timing="90">
          <aside class="notes">
            So one of them is using all the infrastructure we developed within the benchmark for users. At this moment, we already created a basic pipeline for doing trajectory inference, that starts with the guidelines app, uses all the wrappers we used for the benchmark, and then some standard interpretation tools such as manipulation of the trajectory, feature selection and plotting. The plan is to add some functionality on this toolkit, such as integration with RNA velocity, other single-cell tools such as seurat and scanpy, alignment of trajectories, better differential expression, and a modular visualization suite.
          </aside>

          <div class="chapter-heading">
            <span class="chapter-number" style="background-color:#0074D9">6</span> 
            Future of trajectory inference
          </div>
          <div class="container-row chapter-content">
            <div class="full" data-svg="slides/dyno/toolkit.svg"></div>
          </div>
        </section>

        <section data-vertical-align-top id="komparo">
          <aside class="notes">
            I also have a lot of ideas with respect to making benchmarking better. This mainly centers around collaborartive and continuous benchmarking, where you would have several modules, including dataset generators, metrics, methods and report generators, that could be provided by different actors in the field, such as biologists, method developers, and benchmarkers. Combining these modules with modern software technologies, such as containerisation and continuous integration, it would become possible to have a benchmarking platform that is extendable, always up-to-date, reproducible, would embrace disagreement in the field and would avoid double work by developers. In the end, I think this could transform method development in bioinformatics as a whole.
          </aside>


          <div class="chapter-heading">
            <span class="chapter-number" style="background-color:#39ccccff">7</span>
            Future of benchmarking
          </div>
          <div class="container-row chapter-content">
            <div class="full" data-svg="slides/komparo/modules.svg"></div>
          </div>
        </section>


        <section  data-timing="30">
          <aside class="notes">
            So this was my thesis, these were my goals, and although they were of course defined very broad, I hope I still succeeded in them. Thank you.
          </aside>
          <div class="container">
            <div class="col bordered full">
              <div>
                <p>My goals: </p>
                <p>
                  Analyse the transcriptome of <br> macrophages and dendritic cells
                </p>
                <p>
                  But...
                </p>
                <ul class="fa-ul" style="list-style-type: none;padding-left:40px">
                  <li>
                    <span class="fa-li"><i class="far fa-check-square"></i></span>
                    <p>How to do differential expression for 3 conditions?</p>
                  </li>
                  <li>
                    <span class="fa-li"><i class="far fa-check-square"></i></span>
                    <p>So many module detection methods, which one should I use?</p>
                  </li>
                  <li>
                    <span class="fa-li"><i class="far fa-check-square"></i></span>
                    <p>So many trajectory inference detection methods, which one should I use?</p>
                  </li>
                </ul>
              </div>
            </div>
            <div class="col bordered full">
              <div id="overview" data-svg="slides/title/overview.svg"></div>
            </div>
          </div>

        </section>
			</div>
		</div>

		<script src="js/reveal.js"></script>

		<script>
			// More info about config & dependencies:
			// - https://github.com/hakimel/reveal.js#configuration
			// - https://github.com/hakimel/reveal.js#dependencies
			Reveal.initialize({
				dependencies: [
					{ src: 'plugin/markdown/marked.js' },
					{ src: 'plugin/markdown/markdown.js' },
					{ src: 'plugin/notes/notes.js', async: true },
          { src: 'plugin/highlight/highlight.js', async: true },
          {
            src: 'plugin/reveal-svg-fragment.js',
            condition: function(){
              return !!document.querySelector( '[data-svg-fragment]' );
            }
          },
          { src: 'plugin/jquery.js'},
					{ src: 'plugin/reveald3.js'},
					{ src: 'plugin/d3.js'},
					{ src: 'plugin/lodash.js'}
				],
        hash: true,
        width: 1280,
        height: 720,
        transition: 'none',
        transitionSpeed: 'fast',
        fragmentInURL: true,
        defaultTiming: 120,
        controls: false,
        progress: false
      });
    </script>
    <script src="js/fragments.js"></script>
	</body>
</html>
