<!doctype html>
<html>
	<head>
		<meta charset="utf-8">
		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

		<title>Developing & benchmarking methods for analysing transcriptomics data</title>

		<link rel="stylesheet" href="css/reset.css">
		<link rel="stylesheet" href="css/reveal.css">
		<link rel="stylesheet" href="css/theme/solarized.css">
    <link rel="stylesheet" href="css/my_theme.css">
    <link rel="stylesheet" href="fonts/fontawesome.css">

		<!-- Theme used for syntax highlighting of code -->
		<link rel="stylesheet" href="lib/css/monokai.css">

		<!-- Printing and PDF exports -->
		<script>
			var link = document.createElement( 'link' );
			link.rel = 'stylesheet';
			link.type = 'text/css';
			link.href = window.location.search.match( /print-pdf/gi ) ? 'css/print/pdf.css' : 'css/print/paper.css';
			document.getElementsByTagName( 'head' )[0].appendChild( link );
    </script>
	</head>
	<body>
		<div class="reveal">
			<div class="slides">
				<section id="hello">
          <aside class="notes">
            Hello [...]
          </aside>

					<div class="chapter-content">
            <div class="container">
              <div class="col bordered full">
                <div class="container-row">
                  <div class="row">
                    <div>
                      <sup><i class="fas fa-graduation-cap"></i></sup>
                      <h1>Closed PhD defence</h1>
                      Wouter Saelens
                    </div>
                  </div>
                  <div class="row">
                    <div>This presentation:<br> <a href="https://zouter.github.io/closed-defence">zouter.github.io/closed-defence
                        <sup><i class="fas fa-external-link-alt" style="font-size:0.7em"></i></sup></a> </div>
                  </div>
                </div>
              </div>
              <div class="col bordered full">
                <div id="overview" data-svg="slides/title/overview.svg"></div>
              </div>
            </div>
          </div>
        </section>
        

				<section id="big-questions">
          <aside class="notes">
            Why do we even study transcriptomics? Well we have these large questions, such as [...] But to answer them we have to go
            smaller and smaller: look into what cells are doing, which leads us to trying to understand what is happening in this cell. Of course a bunch of molecules, and one molecule that is relatively easy to quantify and which gives us a lot of
            information are the mRNA transcripts. They of course encode for proteins, and their abundances can tell us a lot about
            what the cell is doing.
          </aside>
          <div class="chapter-heading">
            Big questions often have tiny answers
          </div>
          <div class="chapter-content" style="height:500px;">
            <div data-svg="slides/intro/intro.svg" class="full"></div>
          </div>
        </section>

        <section id="intro">
          <aside class="notes">
            Transcriptomics data is highly dimensional, i.e. a couple to thousands of samples with information about all protein coding genes data, and this requires computational tools. Some of those are listed here, starting from the simpler [...] to the more complex [...].

            The goal in my PhD wasn't set in stone from the beginning. Instead I tried to analyze some data in the department and then see where I could contribute new methods, or compare existing methods.
          </aside>

          <div class="container">
            <a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a
              class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a
              class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a
              class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a>
            <div class="col bordered">
              <div data-svg="slides/intro/tools.svg" class="full"></div>
            </div>
            <div class="col bordered" style="line-height:2em;" data-index="goals">
              <div>
                <p>My goals: </p>
                <ul class="fa-ul" style="list-style-type: none;padding-left:40px">
                  <li>
                    <span class="fa-li"><i class="far fa-square"></i></span>
                    Fill the gaps where possible
                  </li>
                  <li>
                    <span class="fa-li"><i class="far fa-square"></i></span>
                    Compare methods where needed
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </section>

        <section data-vertical-align-top id="triwise">
          <aside class="notes">
            One problem that I encountered early in my PhD is the analysis of 3 biological conditions. When we have two conditions, a pairwise analysis is quite easy. Also if you have more samples, some clustering is typically employed to simplify the dataset and visualize it. The problem is, when I had three conditions, the first approach would require you to do many pairwise comparisons making it difficult to interpret. The other approach would also not work that well because the data is too simple to really try to cluster the genes.

            <hr>

            So that's why I developed this visualisation technique where I convert the data to two dimensions using a barycentric transformation. Genes that are not different in any of the three samples are in the middle. Other genes move towards one of the three poles depending on their expression. The distance to the origin is the magnitude of differential expression.
            <br>
            One of the strengths of this analysis is that you can highlight functional gene sets and see whether they are differentialle expressed in a particular direction. For example, here you can see [...]

            <hr>
            Overall, this approach is interesting in that you can use it for many designs, such as the case where you have one reference and two populations of interest. Depending on the biological question, you also have different tests.
            I'm happy with the output that we have with this project, although the approach also has some issues, such as its limitation to three dimensions and the slow implementation of the permutation test. Finally, I think in the future the package could benefit from having a more theoretical statistical background, for example for the self-contained test.
          </aside>

          <a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a>
          <div class="chapter-heading">
            <span class="chapter-number" style="background-color:#FF4136">2</span> 
            From two to three conditions | 
            <span class="chapter-subheading"  style="color:#FF4136">
              <span data-index="problem">
                  problem
              </span>
              <span data-index="solution">
                    solution
              </span>
              <span data-index="conclusion">
                  conclusion
              </span>
            </span>
          </div>

          <div class="chapter-content" data-index="problem">
            <div class = "container">
              <div class="col bordered" data-index="existing">
                <div>
                  <p>Comparing two populations: <br>easy</p>
                  <p><img src="slides/triwise/two_example.png" style="width:75%"></p>
                </div>
              </div>
              <div class="col bordered" data-index="needed">
                <div>
                  <p style="align-self: flex-start">Comparing three populations: </p>
                  <p style="line-height:300px;">a bit more difficult</p>
                </div>   
              </div>
              <div class="col bordered" data-index="existing">
                <div>
                  <p>Comparing many populations: <br>easy</p>
                  <p><img src="slides/triwise/many_example.png" style="width:75%"></p>
                </div>
              </div>
            </div>
          </div>
          <div class="chapter-content" data-index="solution">
            <div class="container">
                <div class="col bordered" data-svg="slides/triwise/lianne.svg" style="flex:2"></div>
                <div class="col bordered" data-svg="slides/triwise/lianne2.svg" style="flex:1" data-index="significant"></div>
            </div>
          </div>

          <div class="chapter-content" data-index="conclusion">
            <div class="container-row conclusion-container">
              <div class="row bordered conclusion-box">
                <div class="conclusion-icon"><i class="fas fa-check"></i></div>
                <div class="conclusion-content">
                  <ul>
                    <li>Useful for multiple designs</li>
                    <li>Different tests depending on the biological question</li>
                  </ul>
                </div>
              </div>
              <div class="row bordered conclusion-box">
                <div class="conclusion-icon"><i class="fas fa-newspaper"></i></div>
                <div class="conclusion-content">
                  <ul>
                    <li>3 co-publications</li>
                    <li>4 other studies</li>
                    <li>R package</li>
                  </ul>
                </div>
              </div>
              <div class="row bordered conclusion-box" data-index="future">
                <div class="conclusion-icon"><i class="fas fa-times"></i></div>
                <div class="conclusion-content">
                  <ul>
                    <li>Limited to 3 dimensions</li>
                    <li>Slow implementation of permutation test</li>
                  </ul>
                </div>
              </div>
              <div class="row bordered conclusion-box" data-index="future">
                <div class="conclusion-icon"><i class="fas fa-tasks"></i></div>
                <div class="conclusion-content">
                  <ul>
                    <li>Better statistical framework</li>
                  </ul>
                </div>
              </div>
            </div>
          </div>


        </section>

        <section data-vertical-align-top id="modbenchmark">
          <aside class="notes">
            My next project concerned the use of gene module detection methods. The reason we want to use these methods is that we want to find genes that are functionally and/or regulatory related. This would help us to better understand the process that we are studying, which is often regulated by many genes. The way we do this is that we first group genes based on their similar expression profile, and we assume that these genes because they are co-expressed, will also share function and regulation.

            There are many methods to find these modules, and my goal was to evaluate them. So what I did was to actually test this assumption: are these methods detecting co-regulated or functionally related modules?

            <hr>

            There are many methods to get modules. The most popular one are of course the clustering methods, such as hierarchical clustering shown here. But you also have alternatives such as decomposition and biclustering, which will look for local co-expression patterns and potentially overlapping modules. You also have network inference based methods which will infer some kind of regulatory network together with module detection.

            


          </aside>


          <a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a>
          <div class="chapter-heading">
            <span class="chapter-number" style="background-color:#FF851B">3</span> 
            Module detection comparison | 
            <span class="chapter-subheading" style="color:#FF851B">
              <span data-index="problem">
                  problem
              </span>
              <span data-index="solution">
                    solution
              </span>
              <span data-index="conclusion">
                  conclusion
              </span>
            </span>
          </div>
          <div class="container-row chapter-content" data-index="problem">
            <div class="full bordered">
              <div data-svg="slides/modbenchmark/assumption.svg" class="full"></div>
            </div>
          </div>

          <div class="container-row chapter-content" data-index="solution">
            <div class="row bordered">
              <div data-svg="slides/modbenchmark/fig2.svg" class="full"></div>
            </div>
          </div>

          <div class="chapter-content" data-index="conclusion">
            <div class="container-row conclusion-container">
              <div class="row bordered conclusion-box">
                <div class="conclusion-icon"><i class="fas fa-check"></i></div>
                <div class="conclusion-content">
                  <ul>
                    <li>First large-scale evaluation of these methods</li>
                    <li>Decomposition methods for local co-expression</li>
                    <li>Pinpoints challenges to interpretation and accuracy</li>
                  </ul>
                </div>
              </div>
              <div class="row bordered conclusion-box">
                <div class="conclusion-icon"><i class="fas fa-newspaper"></i></div>
                <div class="conclusion-content">
                  <ul>
                    <li>1 first author publication</li>
                    <li>Used by developers & users</li>
                  </ul>
                </div>
              </div>
              <div class="row bordered conclusion-box" data-index="future">
                <div class="conclusion-icon"><i class="fas fa-times"></i></div>
                <div class="conclusion-content">
                  <ul>
                    <li>Limited to larger bulk datasets</li>
                    <li>Extendability could have been better</li>
                  </ul>
                </div>
              </div>
              <div class="row bordered conclusion-box" data-index="future">
                <div class="conclusion-icon"><i class="fas fa-tasks"></i></div>
                <div class="conclusion-content">
                  <ul>
                    <li>Single-cell data changes the playing field</li>
                    <li>Other omics data types</li>
                  </ul>
                </div>
              </div>
            </div>
          </div>


          <aside class="notes">
            What we want to know is which genes share regulation (ie. to find their common regulator that can be targeted) or function (i.e. to understand more about the functional process)
          </aside>
        </section>




        <section data-vertical-align-top id="dynbenchmark">
          <a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a
            class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a
            class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a
            class="fragment"></a><a class="fragment"></a><a class="fragment"></a><a class="fragment"></a>
          <div class="chapter-heading">
            <span class="chapter-number" style="background-color:#93cc2e">4</span>
            Trajectory inference comparison |
            <span class="chapter-subheading" style="color:#93cc2e">
              <span data-index="problem">
                problem
              </span>
              <span data-index="solution">
                solution
              </span>
              <span data-index="conclusion">
                conclusion
              </span>
            </span>
          </div>

          <div class="container-row chapter-content bordered" data-index="problem-1">
              <div class="full" data-svg="slides/dynbenchmark/trajectory_example.svg"></div>
          </div>

          <div class="container-row chapter-content bordered" data-index="problem-2">
              <div class="full" data-svg="slides/dynbenchmark/history.svg"></div>
          </div>
 
        
          <div class="container-row chapter-content bordered" data-index="solution-1">
            <div class="full" data-svg="slides/dynbenchmark/guidelines.svg"></div>
          </div>

          <div class="container-row chapter-content bordered" data-index="solution-2">
            <iframe src="slides/dynbenchmark/dynguidelines.html" style="width:100%;height:100%;max-width:100%;max-height:100%;"></iframe>
          </div>
        
          <div class="chapter-content" data-index="conclusion">
            <div class="container-row conclusion-container">
              <div class="row bordered conclusion-box">
                <div class="conclusion-icon"><i class="fas fa-check"></i></div>
                <div class="conclusion-content">
                  <ul>
                    <li>First large-scale evaluation of these methods</li>
                    <li>Interactive context-dependent guidelines</li>
                  </ul>
                </div>
              </div>
              <div class="row bordered conclusion-box">
                <div class="conclusion-icon"><i class="fas fa-newspaper"></i></div>
                <div class="conclusion-content">
                  <ul>
                    <li>2 shared first-author publication</li>
                    <li>3 preprints</li>
                    <li>Dozens of R packages</li>
                  </ul>
                </div>
              </div>
              <div class="row bordered conclusion-box" data-index="future">
                <div class="conclusion-icon"><i class="fas fa-times"></i></div>
                <div class="conclusion-content">
                  <ul>
                    <li>Problematic extendability</li>
                    <li>Already outdated</li>
                    <li>Some discussion in the field</li>
                  </ul>
                </div>
              </div>
              <div class="row bordered conclusion-box" data-index="future">
                <div class="conclusion-icon"><i class="fas fa-tasks"></i></div>
                <div class="conclusion-content">
                  <ul>
                    <li>Crowd-sourced competition</li>
                    <li>Countless extensions are possible</li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        
        
          <aside class="notes">

          </aside>
        </section>


        <section data-vertical-align-top>
          <div class="chapter-heading">
            <span class="chapter-number" style="background-color:darkgray">5</span>
            Overall conclusions
          </div>
          <div class="container-row" style="font-size:0.8em">
            <div class="row bordered" >
              Benchmarking is important for
              <ul style="list-style: none;text-align:left;">
                <li>
                  → users, because popular ⇏ best performance
                </li>
                <li>
                  → developers, because good ideas ⇏ better for users
                </li>
              </ul>
            </div>
            <div class="row bordered">
              Accuracy, scalability, usability and stability are not always correlated. <br>
              Find a balance
            </div>
            <div class="row bordered">
               Performance is context-dependent <br>
               Avoid falling in the "ranking" trap
            </div>
            <div class="row bordered">
              Benchmarking studies can easily contradict <br>
              Finding common ground is difficult
            </div>
          </div>
        </section>

        <section data-vertical-align-top id="dyno">
          <div class="chapter-heading">
            <span class="chapter-number" style="background-color:#0074D9">6</span> 
            Future of trajectory inference
          </div>
          <div class="container-row chapter-content">
            <div class="full" data-svg="slides/dyno/toolkit.svg"></div>
          </div>
        </section>

        <section data-vertical-align-top id="komparo">
          <div class="chapter-heading">
            <span class="chapter-number" style="background-color:#39ccccff">7</span>
            Future of benchmarking
          </div>
          <div class="container-row chapter-content">
            <div class="full" data-svg="slides/komparo/modules.svg"></div>
          </div>
        </section>


        <section>
          <div class="container">
            <div class="col bordered full">
              <div id="overview" data-svg="slides/title/overview.svg"></div>
            </div>
            <div class="col bordered full">
              <div style="line-height:2em;">
                My goals:
                <ul class="fa-ul" style="list-style-type: none;padding-left:40px">
                  <li>
                    <span class="fa-li"><i class="far fa-check-square"></i></span>
                    Fill the gaps where possible
                  </li>
                  <li>
                    <span class="fa-li"><i class="far fa-check-square"></i></span>
                    Compare methods where needed
                  </li>
                </ul>
              </div>
            </div>
          </div>

        </section>
			</div>
		</div>

		<script src="js/reveal.js"></script>

		<script>
			// More info about config & dependencies:
			// - https://github.com/hakimel/reveal.js#configuration
			// - https://github.com/hakimel/reveal.js#dependencies
			Reveal.initialize({
				dependencies: [
					{ src: 'plugin/markdown/marked.js' },
					{ src: 'plugin/markdown/markdown.js' },
					{ src: 'plugin/notes/notes.js', async: true },
          { src: 'plugin/highlight/highlight.js', async: true },
          {
            src: 'plugin/reveal-svg-fragment.js',
            condition: function(){
              return !!document.querySelector( '[data-svg-fragment]' );
            }
          },
          { src: 'plugin/jquery.js'},
					{ src: 'plugin/reveald3.js'},
					{ src: 'plugin/d3.js'},
					{ src: 'plugin/lodash.js'}
				],
        hash: true,
        width: 1280,
        height: 720,
        transition: 'none',
        transitionSpeed: 'fast',
        fragmentInURL: true,
        defaultTiming: 120
      });
    </script>
    <script src="js/fragments.js"></script>
	</body>
</html>
